### PR TITLE
Persistence layer interface

### PR DESCRIPTION
**Pull request by daltonfury42**
Define the interface of persistence layer. We would have one mongo based implementation for now. 

Writing the Mongo stuff behind an interface would help us support multiple data storage systems, say for example if we feel tomorrow that we need to migrate to DynamoDB or install inhouse for a client who wants to store data in PostgreSQL. Also this will help us easily mock out the DB layer for testing our business logic.

### CODE REVIEW

File: internal/models/queue.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	QueueName string
+
+	// Current status of the queue.
+	Status QueueStatus
+
+	// Tokens present in the queue.
+	Tokens []Token
+
+	// Creation time.
+	creationTime time.Time
```
Comment (maaverik) [inline internal/models/queue.go]: There are camel case and pascal case variable names. Can we stick with any one convention?

Comment (daltonfury42) [reply internal/models/queue.go]: Ah. Good catch.  This should have been PascalCase. 

Golang enforces that public references are Pascal and private ones would be camel.

**Review by aneeshsharma — APPROVED** [review]

File: internal/persistence/persistence.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package persistence
+
+import (
+	"github.com/SimplQ/simplQ-golang/internal/models"
+)
+
+type QueueStore interface {
+	// Create a new queue and return the queue ID.
+	CreateQueue(models.Queue)
```
Comment (aneeshsharma) [inline internal/persistence/persistence.go]: This should have a return type as `Id` I think

**Review by aneeshsharma — CHANGES_REQUESTED** [review]

File: internal/models/queue.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	// Unique ID for the queue. 
+	Id QueueId
+
+	// Name of the queue. The name is used in generating queue urls, for ex 
+	// https://simplq.me/j/<QueueName>. The storage layer guarantees sure that 
+	// there is only one queue by a given name.
+	QueueName string
+
+	// Current status of the queue.
+	Status QueueStatus
```
Comment (aneeshsharma) [inline internal/models/queue.go]: Pause and delete status should be represented by 2 different fields

**Review by aneeshsharma — APPROVED** [review]

**Pull request merged**